homeassistant/components/mqtt/client.py

```diff
@@ -893,14 +893,35 @@ async def _async_perform_subscriptions(self) -> None:
         if not self._pending_subscriptions:
             return
 
-        subscriptions: dict[str, int] = self._pending_subscriptions
+        pending_subscriptions: dict[str, int] = self._pending_subscriptions
+        pending_wildcard_subscriptions = {
+            subscription.topic: pending_subscriptions.pop(subscription.topic)
+            for subscription in self._wildcard_subscriptions
+            if subscription.topic in pending_subscriptions
+        }
+        simple_subscription_list = list(pending_subscriptions.items())
```

Comment:
This is an existing issue, but creating a list here seems useless. Make an iterator instead.

Reply:
Removed the overhead